### PR TITLE
Remove pointers from slices

### LINKED ISSUE #376: Get rid of the overhead use of pointers in field setters and `repeated` fields
Currently we use pointers in `nested` and `repeated` message structures. With seemingly useful savings in stack memory, we thus increase the cost of garbage collection, because binding by pointer loads the heap. From the other side we use pointers where there is no memory modification, which is wrong ideologically. I propose to work with nested/repeated messages by value everywhere.

Additionally we can somewhere reduce the number of message types by flattening top-level messages when their nested structures they are not usable by themselves (e.g. `body` messages).

### PR DESCRIPTION
**Pull request by fyrchik**
Close #376 . Depends on #377 .

### CODE REVIEW

File: reputation/grpc/service.pb.go [hunk 1]
```diff
@@ -7,11 +7,12 @@
 package reputation
 
 import (
+	reflect "reflect"
+	sync "sync"
+
```
Comment (alexvanin) [inline reputation/grpc/service.pb.go:12]: Not sure this should have been changed :thinking:

File: session/grpc/types.pb.go [hunk 2]
```diff
@@ -7,13 +7,14 @@
 package session
 
 import (
+	reflect "reflect"
+	sync "sync"
+
```
Comment (alexvanin) [inline session/grpc/types.pb.go:12]: One more pb.go file change.

**Review by alexvanin — APPROVED** [review]

File: reputation/grpc/service.pb.go (hunk 1 above)
Comment (fyrchik) [reply reputation/grpc/service.pb.go]: Fixed.

File: session/grpc/types.pb.go (hunk 2 above)
Comment (fyrchik) [reply session/grpc/types.pb.go]: Fixed.

**Review by alexvanin — APPROVED** [review]

**Review by carpawell — APPROVED** [review]

**Comment by alexvanin** [comment]
Got some issues with adopting it in SDK (netmap package), do not merge yet.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**